src/flint/types/fmpz_mod.pyx

```diff
@@ -126,11 +126,11 @@ cdef class fmpz_mod_ctx:
         # If they're not the same object in memory, they may have the
         # same modulus, which is good enough
         if typecheck(other, fmpz_mod_ctx):
-            return fmpz_equal(self.val.n, (<fmpz_mod_ctx>other).val.n)
+            return fmpz_equal(self.val.n, (<fmpz_mod_ctx>other).val.n) == 1
         return False
 
     def __hash__(self):
-        return hash(self.modulus())
+        return hash(repr(self))
```

Comment:
I don't think that this is necessary. I expect that computing the repr here could sometimes be very expensive compared to hashing the modulus.

Reply:
OK, I'll revert this